### ISSUE TITLE
How big file?

### ISSUE THREAD

**Issue by rizplate** [RaRe-Technologies/smart_open#179]
1. How big a file can it upload from user's client machine to server to s3? 1G, 500G, 1T?
1. Can it upload binary files
1. Is there a test that shows upload from requests to webserver i.e. flask to s3?

**Comment by mpenkov** [comment]
1. The theoretical limit of the size of an S3 object is currently 5TB. As for the practical limit when using smart_open, I don't think have any tests for file size, do we @menshikh-iv?
2. Yes
3. I'm not sure I understand the question. The tests are here, please have a look: https://github.com/RaRe-Technologies/smart_open/tree/master/smart_open/tests

**Comment by menshikh-iv** [comment]
@mpenkov we have no tests for it, but I don't see even potential problems with large files.
@rizplate can you rephrase your question please, I don't catch it too.

**Comment by rizplate** [comment]
@menshikh-iv I meant that can a webserver be added in between? i.e. a client uploads to a webserver using http post and the webserver then uploads to the s3 using smart_open. The particular webserver I am talking about is flask or django.

**Comment by mpenkov** [comment]
@rizplate That doesn't sound like a test, that sounds like an integration with a Web framework. You would have to install smart_open with its dependencies on the Web server. You would also have to write the integration code, which reads the file uploaded to the server, and writes it to S3 via smart_open.
